src/oid.rs

```diff
@@ -79,6 +80,14 @@ impl Default for ObjectId {
     }
 }
 
+impl FromStr for ObjectId {
+    type Err = Error;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
```

Comment:
Would you mind adding a test to ensure this method works as expected? Feel free to refer to the tests in `src/tests/modules/oid.rs` as a guide.